Backend "Tydzień 1" zadania 1 and 2

lyneq2001 opened the pull request.

lyneq2001 committed:
Backend "Tydzień 1" zadania 1 and 2

nxtkofi reviewed (changes requested):
Jest już serwer postawiony razem z bazą w repozytorium. do przepisania w Javie w /server + baza danych jest w /database

lyneq2001 committed:
Dodanie Backend TaskMaster

lyneq2001 committed:
dodanie backend TaskMaster

nxtkofi reviewed (changes requested):
pisałem Ci w poprzednim PR, że backend już jest postawiony w folderze ./server/
Michał robił tam zmiany no więc Ty też tam powinieneś to implementować. Po co nam 3 osobne serwery razem z Twoimi? Jeden zrobiłeś nowy folder w Backend/taskmanager a drugi postawiłeś obok w pythonie.
To po pierwsze a po drugie miałeś zająć się **TYLKO**  autoryzacją.
Przeczytaj issue: https://github.com/nxtkofi/TaskMasters/issues/18
Michał zajął się już postawieniem całej reszty, autoryzacja pozostała dla Ciebie. Do poprawki to będzie, ściągnij zmiany z maina, powinny tam być już zmiany Michała i pracuj na tym co on zrobił czyli dodaj tam to co masz w tickecie.
No i usuń ten serwer pythona bo już ustaliliśmy, że ma być w Javie

lyneq2001 committed:
Merge branch 'master' into build/20-backend-week1

lyneq2001 committed:
build/18-authorization

lyneq2001 committed:
usuniecie folderu Backend

lyneq2001 committed:
poprawa plików

lyneq2001 committed:
chenge of Lombok version to 1.18.38, added user login, user registration (with password hashing) and Authentication with JWT tokens

nxtkofi commented on TasksAndTeams.py:
Miałeś usunąć, skoro nie korzystamy z pythona

nxtkofi commented on authentication.py:
To samo co wyżej.

nxtkofi commented on server/pom.xml:
po co komentarze?

nxtkofi commented on server/src/main/java/pl/server/Config/JwtUtility.java:
```diff
@@ -0,0 +1,65 @@
+@Component
+public class JwtUtility {
+
+    private final String SECRET_KEY = "your-256-bit-secret-key-1234567890abcdef"; // Replace with a secure key
```
to do application.properties powinno isc i tutaj wtedy uzywasz @Value zeby tego uzyc

nxtkofi commented on server/src/main/java/pl/server/Config/JwtUtility.java:
```diff
@@ -0,0 +1,65 @@
+public class JwtUtility {
+
+    private final String SECRET_KEY = "your-256-bit-secret-key-1234567890abcdef"; // Replace with a secure key
+    private final long JWT_EXPIRATION = 1000 * 60 * 60 * 10; // 10 hours
```
to też do application.properties

nxtkofi commented on server/src/main/java/pl/server/Config/SecurityConfig.java:
```diff
@@ -2,30 +2,51 @@
 
+    private final JwtRequestFilter jwtRequestFilter;
+
+    // Użycie @Lazy, aby przerwać cykl zależności
```
kom do usuniecia

nxtkofi commented on server/src/main/java/pl/server/Controllers/AuthController.java:
```diff
@@ -1,20 +1,52 @@
+
     @PostMapping("/register")
     public ResponseEntity<String> register(@RequestBody User user) {
+        user.setRole(UserRole.CEO); // Default role for new users (as per demo scenario)
```
nie chcemy zeby kazdy user był domyślnie CEO. W tym enumie jest lepsza rola dla userów którzy dopiero co się zarejestrowali

nxtkofi replied:
kom do usuniecia

nxtkofi commented on server/src/main/java/pl/server/Controllers/LoginRequest.java:
```diff
@@ -0,0 +1,37 @@
+import lombok.Setter;
+import lombok.ToString;
+
+@ToString
+@Data
+@Setter
+@EqualsAndHashCode
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
```
z tego co pamiętam to `@Data` implementuje wszystko co tutaj wypisałeś więc raczej wszystko do usunięcia poza `@Data`

nxtkofi commented on server/src/main/java/pl/server/Controllers/LoginRequest.java:
```diff
@@ -0,0 +1,37 @@
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
```
Trzeci raz definiujesz gettery i settery - raz w `@Getter` i `@Setter`, drugi w `@Data` i trzeci raz tutaj hardcoded. Do usuniecia

nxtkofi commented on server/src/main/java/pl/server/Controllers/LoginRequest.java:
Plik w złym miejscu, takie rzeczy wrzuć do folderu Dtos

nxtkofi commented on server/src/main/java/pl/server/Entity/User.java:
to są stare rzeczy z brancha michała - czemu  one są tutaj?

nxtkofi commented on server/src/main/java/pl/server/Service/UserService.java:
```diff
@@ -1,25 +1,47 @@
+        this.passwordEncoder = passwordEncoder;
+    }
 
     public User registerUser(User user) {
+        if (user == null || user.getEmail() == null || user.getPassword() == null) {
+            throw new IllegalArgumentException("User, email, or password must not be null");
+        }
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         return userRepository.save(user);
     }
```
Ogólnie to nie było do końca Twoje zadanie na ten ticket, ale jak już się za to zabrałeś to warto byłoby sprawdzić, czy user o tym samym email nie istnieje już w naszej bazie

nxtkofi commented on server/src/test/java/pl/server/ServerApplicationTests.java:
po co konfigurujesz testy? To wykracza poza Twoje zadania w tym tickecie, nie planujemy testów narazie

nxtkofi commented on server/src/main/resources/application-test.properties:
jezeli robisz sobie jakies zmiany do testów to nie pusuhj ich w commicie

nxtkofi commented on server/src/main/resources/application.properties:
te zmiany nie są potrzebne, application.properties z mastera działa w zupełności dobrze

lyneq2001 committed:
Poprawki w projekcie, żeby był bardziej przejrzysty i bezpieczny:

- Wywalono cały kod Pythona i komentarze z pom.xml, żeby był porządek.
- JwtUtility przerobione, teraz ciągnie SECRET_KEY i JWT_EXPIRATION z application.properties przez @Value.
- W AuthController zmieniono domyślną rolę dla nowych użytkowników na USER, a nie CEO w metodzie register.
- Uproszczono LoginRequest: wyrzucono niepotrzebne adnotacje Lombok (@ToString, @Setter, @EqualsAndHashCode, @Getter) i ręczne gettery/settery, zostawiono @Data, @AllArgsConstructor i @NoArgsConstructor.
- Dodano sprawdzanie w UserService, czy e-mail już istnieje w registerUser – jak jest zajęty, to leci wyjątek IllegalArgumentException.
- Przywrócono application.properties i ServerApplicationTests.java do oryginalnego stanu.
- Stworzono folder Dtos i wrzucono tam LoginRequest dla lepszej organizacji.
